typegen/strats/inline.py

```diff
@@ -1,96 +1,110 @@
-import ast
 import functools
 import logging
 import operator
 import pathlib
-import itertools
+
+import pandas as pd
+
+import libcst as cst
+
+from libcst.metadata import PositionProvider
 
 from constants import TraceData
 from tracing.trace_data_category import TraceDataCategory
 from typegen.strats.gen import TypeHintGenerator
 
-import pandas as pd
-
 logger = logging.getLogger(__name__)
 
 
-class TypeHintTransformer(ast.NodeTransformer):
-    def __init__(self, relevant: pd.DataFrame) -> None:
-        super().__init__()
-        self.df = relevant
+class TargetExtractor(cst.CSTVisitor):
+    targets: list[tuple[str, cst.Name | cst.Attribute]]
```

Comment:
What is the usage of this member?

Reply:
It is used to extract to store the targets on the LHS of an assignment.
This class is needed because the new lib does not offer a counterpart to `ast.walk` or `ast.iter_child_nodes`

Reply:
I mean the targets variable, as the instances already contain a targets member.

Reply:
Ah ok, it is for type hinting.